### PR TITLE
refactor(fiscal): best practices for closing

### PR DESCRIPTION
**Pull request by jniles**
This commit updates the Fiscal Year module with the current best coding
practices.  It specifically targets the closing fiscal year code and
updates it with the following changes:
 1. The translate directive is used instead of the translate filter
 2. The bhAccountSelect is used instead of a custom ui-select
 3. The modal uses a headercrumb(tm)
 4. Removed bhFormDefaults from the form.

This commit addresses most of the points in #1646 for closing a fiscal
year.  Tests have been updated accordingly. The rest of the fiscal year
modules should be updated in a similar fashion.

----

Thank you for contributing!

Before submitting this pull request, please verify that you have:
 - [x] Run your code through ESLint.  [Check out our styleguide](https://github.com/IMA-WorldHealth/bhima-2.X/blob/master/docs/STYLEGUIDE.md).
 - [x] Run integration tests.
 - [x] Run end-to-end tests.
 - [x] Accurately described the changes your are making in this pull request.

For a more detailed checklist, [see the official review checklist](https://docs.google.com/document/d/1nupLVLRXgSZJQo_acLgrwvPnN8RukfSiwRhSToj81uU/pub) that this PR will be evaluated against.

Ensuring that the above checkboxes are completed will help speed the review process and help build a stronger application.  Thanks!

### CODE REVIEW

**Comment by jniles** [comment]
@mbayopanda, can you give me a review on this PR?

**Comment by sfount** [comment]
@jniles I will review this code as soon as possible today.

File: client/src/modules/fiscal/templates/modals/fiscal.closing.modal.html [hunk 1]
```diff
@@ -1,111 +1,90 @@
-<form name="ClosingFiscalYear" bh-submit="$ctrl.stepForward(ClosingFiscalYear)" bh-form-defaults novalidate>
-  <div class="modal-header bg-danger text-danger">
-    <i class="fa fa-lock"></i>
-    <strong>{{ 'FISCAL.CLOSING_FISCAL_YEAR' | translate }}</strong>
+<form name="ClosingFiscalYear" bh-submit="$ctrl.stepForward(ClosingFiscalYear)" novalidate>
+  <div class="modal-header">
+    <ol class="headercrumb">
```
Comment (sfount) [inline client/src/modules/fiscal/templates/modals/fiscal.closing.modal.html:3]: :tm:

File: client/src/modules/fiscal/templates/modals/fiscal.closing.modal.html [hunk 2]
```diff
@@ -1,111 +1,90 @@
...
-                <span ng-bind-html="account.label | highlight:$select.search"></span>
-              </ui-select-choices>
-            </ui-select>
-
-            <div class="help-block" ng-messages="ClosingFiscalYear.resultAccount.$error" ng-show="ClosingFiscalYear.$submitted">
-              <div ng-messages-include="modules/templates/messages.tmpl.html"></div>
-            </div>
-          </div>
-
+          <bh-account-select
```
Comment (sfount) [inline client/src/modules/fiscal/templates/modals/fiscal.closing.modal.html:50]: 👍

**Review by sfount — APPROVED** [review]
This is a great re-factor. Code LGTM.

**Pull request merged**